src/main/java/luoyi/duke/data/Duke.java

```diff
@@ -0,0 +1,122 @@
+package luoyi.duke.data;
+
+import luoyi.duke.commands.Command;
+import luoyi.duke.common.Message;
+import luoyi.duke.common.TextFormatter;
+import luoyi.duke.data.exception.DukeIllegalArgumentException;
+import luoyi.duke.data.exception.DukeUnrecognizedArgumentException;
+import luoyi.duke.data.task.ITask;
+import luoyi.duke.data.task.TaskList;
+import luoyi.duke.parser.Parser;
+import luoyi.duke.storage.Storage;
+import luoyi.duke.ui.Ui;
+
+import java.util.Scanner;
+
+/**
+ * Duke chatbot class to encapsulate the behavior of the chatbot.
+ * Task id starts from 1.
+ */
+public class Duke implements IDuke {
+    /** List for storing Tasks */
+    private final TaskList list;
```

Comment:
naming it list can be ambiguous, maybe it can be called taskList to make it clearer